spec/features/supply_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'rails_helper'
+
+describe "supply", type: :feature, js: true do
+
+  let(:circle) { create(:circle) }
+
+  let!(:circle_admin_role)      { create :circle_role_admin, circle: circle }
+
+  let(:user_1) { create(:user, primary_circle: circle) } #Admin
+  let(:user_2) { create(:user, primary_circle: circle) } #Volunteer
+
+  let(:public_group)  { create(:working_group, is_private: false, circle: circle) }
+  let(:private_group) { create(:working_group, is_private: true , circle: circle) }
+
+  def create_supply(name, description, due_date, location)
```

Comment:
How about using a supply factory here, and only passing in the attributes that should be overridden? That's the usual pattern in rspec tests that require setup data. It would look something like this:

def create_supply(overrides = {})
  # this will use the Factory found in factories/supply.rb and pass all overrides handed into the method
  attrs = attributes_for(:supply, overrides)
  fill_in "Name of supply", with attrs.name
  [...]
end

I'll continue the comment down below where the factory is used.